Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.cpp

```diff
@@ -6,208 +6,107 @@
-        static AZ::EnvironmentVariable<AutoGenRegistryManager> g_autogenRegistry;
-
-        // Look up variable in AZ::Environment first
-        // This is need if the Environment variable was already created
-        if (!g_autogenRegistry)
-        {
-            g_autogenRegistry = AZ::Environment::FindVariable<AutoGenRegistryManager>(ScriptCanvasAutoGenRegistryName);
-        }
-
-        // Create the environment variable in O3DEKernel memory space if it has not been found
-        if (!g_autogenRegistry)
-        {
-            g_autogenRegistry = AZ::Environment::CreateVariable<AutoGenRegistryManager>(ScriptCanvasAutoGenRegistryName);
-        }
-
-        return &(g_autogenRegistry.Get());
+        descriptor->ReleaseDescriptor();
     }
 
-    AZStd::vector<AZStd::string> AutoGenRegistryManager::GetRegistryNames(const char* registryName)
-    {
-        AZStd::vector<AZStd::string> result;
-        result.push_back(AZStd::string::format("%s%s", registryName, ScriptCanvasAutoGenDataRegistrySuffix).c_str());
-        result.push_back(AZStd::string::format("%s%s", registryName, ScriptCanvasAutoGenFunctionRegistrySuffix).c_str());
-        result.push_back(AZStd::string::format("%s%s", registryName, ScriptCanvasAutoGenNodeableRegistrySuffix).c_str());
-        result.push_back(AZStd::string::format("%s%s", registryName, ScriptCanvasAutoGenGrammarRegistrySuffix).c_str());
-        return result;
-    }
+    m_descriptors.clear();
+    m_registeredReflections.clear();
+}
 
-    void AutoGenRegistryManager::Init()
+void ScriptCanvasModel::Init()
+{
+    for (auto descriptor : m_descriptors)
     {
-        auto registry = GetInstance();
-        auto nodeRegistry = NodeRegistry::GetInstance();
-        if (registry && nodeRegistry)
+        if (auto componentApplication = AZ::Interface<AZ::ComponentApplicationRequests>::Get())
         {
-            for (auto& iter : registry->m_registries)
-            {
-                if (iter.second)
-                {
-                    iter.second->Init(nodeRegistry);
-                }
-            }
-        }
-    }
+            componentApplication->RegisterComponentDescriptor(descriptor);
 
-    void AutoGenRegistryManager::Init(const char* registryName)
-    {
-        auto registry = GetInstance();
-        auto nodeRegistry = NodeRegistry::GetInstance();
-        if (registry && nodeRegistry)
-        {
-            auto registryNames = registry->GetRegistryNames(registryName);
-            for (auto name : registryNames)
+            if (m_verbose)
             {
-                auto registryIter = registry->m_registries.find(name.c_str());
-                if (registryIter != registry->m_registries.end())
-                {
-                    registryIter->second->Init(nodeRegistry);
-                }
+                AZ_TracePrintf("ScriptCanvas", "Register Descriptor: %s\n", descriptor->GetName());
```

Comment:
Nit: Should be AZ_Info. Could also be AZ_Trace, but since it's wrapped in the `if (m_verbose)`, I think AZ_Info is probably more reasonable here. It also wouldn't need the \n any more if changed.